src/ripple/overlay/impl/Handshake.cpp

```diff
@@ -288,8 +289,8 @@ verifyHandshake(
     PublicKey const publicKey = [&headers] {
         if (auto const iter = headers.find("Public-Key"); iter != headers.end())
         {
-            auto pk =
-                parseBase58<PublicKey>(TokenType::NodePublic, iter->value());
+            auto pk = parseBase58<PublicKey>(
```

Comment:
This would entail changing multiple functions along the stack until `decodeBase58Token` in `tokens.cpp`. I'm concerned about my earlier comment about the lifetime of the `std::string_view` object, otherwise I'm happy to do this change.

Reply:
fixed in https://github.com/XRPLF/rippled/pull/4509/commits/2cf0eb896c64d078133f108fc1fe55d79c349391
On a related note, if I add a `Json::Value::asStringView()` function overload, would it provide meaningful benefit? It will reduce one copy of the string object

Reply:
A `std::string` is already implicitly convertible to a `std::string_view`, so I don't think an `asStringView()` method is valuable. I do think an `operator= (std::string_view const&)` overload would be valuable, however.

Reply:
`Json::Value::asString()` returns a copy of a string object. If we could have a `asStringView()` function, we will not need to perform such a copy. The strings in an incoming request would not need an extra copy, if we use an `asStringView` function.

Is `operator= (std::string_view const&)` needed? I'm wondering if passing a `std::string const&` or `std::string_view const&` is better, for the function parameter type?

`operator= (std::string_view const&)` takes up 132 lines 
`operator= (std::string const&)` takes up 122 lines 

From a performance point of view, is `std::string_view const&` a lot better than `std::string const&`?

https://godbolt.org/z/MM1rG3fhj

Reply:
In my opinion, `Json::Value::asString()` should return a `std::string_view`. Let callers handle the case where they need to copy the data.

For `operator=`, when callers already have a `std::string`, passing `std::string const&` is best, basically free. Passing `std::string_view const&` requires constructing and destroying a temporary object, but it is small and allocated entirely on the stack, and it can be optimized away if the compiler knows enough about the callee. But when callers only have a `std::string_view`, which many will after this change, they have to construct and destroy a `std::string`, which requires a heap allocation and deallocation, and that is expensive. Further, that construction must be explicit in the code, which is inconvenient for developers to write and read. Thus, both overloads should be available.

Reply:
that makes sense, yes